src/sql/src/pure.rs

```diff
@@ -439,31 +451,87 @@ pub struct Schema {
     pub confluent_wire_format: bool,
 }
 
+#[derive(Debug)]
+pub enum GetSchemaError {
+    Subject(GetBySubjectError),
+    Id(GetByIdError),
+}
+
+impl From<GetBySubjectError> for GetSchemaError {
+    fn from(inner: GetBySubjectError) -> Self {
+        Self::Subject(inner)
+    }
+}
+
+impl From<GetByIdError> for GetSchemaError {
+    fn from(inner: GetByIdError) -> Self {
+        Self::Id(inner)
+    }
+}
+
+impl std::fmt::Display for GetSchemaError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            GetSchemaError::Subject(e) => write!(f, "failed to look up schema by subject: {e}"),
+            GetSchemaError::Id(e) => write!(f, "failed to look up schema by id: {e}"),
+        }
+    }
+}
+
+impl StdError for GetSchemaError {
+    fn source(&self) -> Option<&(dyn StdError + 'static)> {
+        match self {
+            GetSchemaError::Subject(e) => Some(e),
+            GetSchemaError::Id(e) => Some(e),
+        }
+    }
+}
```

Comment:
I don't see how that is clearer. It requires learning a separate DSL just to write the type.

I mean, I could also write a Python script that generates this code, and it would also be short, but I prefer to write Rust?